TestDataCleaning.py

```diff
@@ -0,0 +1,34 @@
+import unittest
+from pandas import DataFrame
+from RealEstateDataCleaning import add_upper_and_main
+
+class TestDataCleaning(unittest.TestCase):
+
+    def check_room_count(self, frame, pk, expected):
+        row = frame.loc[pk]
+
+        # check that the two are added correctly
+        self.assertEqual(expected, row['Hallways'], "Wrong value for 'Hallways' in row {}".format(pk))
+
+        # also ensure that kitchens were unaffected
+        self.assertEqual(1, row['MainKitchens'], "Wrong value for 'MainKitchens' in row {}".format(pk))
+
+    def test_add_upper_and_main(self):
+        data = {
+            'MainHallways': [0, 1, 1],
+            'UpperHallways': [0, 0, 1],
+            'MainKitchens': [1, 1, 1]
+        }
+
+        frame = DataFrame(data, index=[0, 1, 2])
+        add_upper_and_main(frame, 'Hallways')
+
+        # should have 'Hallways', but no 'Main' or 'UpperHallways'
+        self.assertIsNotNone(frame.get('Hallways', None), "'Hallways' should exist")
```

Comment:
So, tests.  Get used to writing tests.  They are going to make your life better.

You should write tests as you write code.  Code a little, test a little.  This will make you write code in a different, almost always better, way.  This will also give you confidence that your code is mostly working the entire time you are working on it.  This will also give you confidence that you didn't unintentionally break something :)

The `unittest` framework that ships with Python is based out of the JUnit family tree.  There are other ones (such as BDD and rspec type things), but they all have the same basic workflow

- Set up some state
- Run some code
- Assert that some things are now true or false.

In this code, I create a simple data frame that has main hallways, upper hallways, and main kitchens.  I run my new `add_upper_and_main` functions.

I then make some broad assertions about the data frame.  I should see a new column called 'Hallways'.  I should no longer see the 'MainHallways' and 'UpperHallways' columns.

After that, I check some individual sets of data.  I want to ensure that the main and upper values were added together correctly.  I also want to ensure that I'm not leaking into other data, so I check to ensure that "MainKitchens" has been unchanged.

You can try changing some of the assert values here to see what a failing test looks like.  There are a few dozen assert functions, from asserting equality to greater than/less than, to "something is in a list", and on and on.